tools/linter/clang_tidy/max_tokens_pragma.py

```diff
@@ -0,0 +1,101 @@
+import argparse
+import re
+from typing import List
+
+
+DEFAULT_MAX_TOKEN_COUNT = 1
```

Comment:
what's the intuition for this being 1?

Reply:
`clang-tidy` doesn't have a direct way to query for token counts in the codebase. The workaround is to set the max-token count to 1, and then `clang-tidy` will output the actual number of tokens for the corresponding file in the error message. 

A non-destructive way to do this would be to pass `-fMaxTokens` through the command line when invoking `clang-tidy`. However, the semantics dictate that the value passed in to `-fMaxTokens` will be overriden if there exists a `#pragma max_tokens_total` statement. Therefore, we need an easy way to set the pragma token counts to 1.

This is why I chose the default to be 1. Might be a good idea to document this.

Reply:
ok, thanks!

> Might be a good idea to document this.

yeah if you could add that in a code comment here, that would be awesome :)